arch/arm/src/xmc4/xmc4_pwm.c

```diff
@@ -0,0 +1,1364 @@
+        }
+    }
+  else
+    {
+      uint32_t scu_ccu4x_mask = (uint32_t)(1 << (priv->module + 2));
+
+#ifdef XMC4_SCU_GATING
+      if ((getreg32(XMC4_SCU_CGATSTAT0) && scu_ccu4x_mask))
+        {
+          putreg32(scu_ccu4x_mask, XMC4_SCU_CGATCLR0);
+        }
+
+#endif
+      if ((getreg32(XMC4_SCU_PRSTAT0) && scu_ccu4x_mask))
+        {
+          putreg32(scu_ccu4x_mask, XMC4_SCU_PRCLR0);
+        }
+    }
+
+  pwminfo("PWM CCU4%d module is enabled\n", priv->module);
+
+  return OK;
+}
+
+/*****************************************************************************
+ * Name: pwm_enable_module
+ *
+ * Description:
+ *   Disable the CCU4x module (gate and assert reset).
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ *****************************************************************************/
+
+static int pwm_disable_module(struct xmc4_pwm_s *priv)
+{
+  DEBUGASSERT(priv != NULL);
+
+  if (priv->module == 3)
+    {
+      /* Assert reset */
+
+      putreg32(SCU_PR1_CCU43RS, XMC4_SCU_PRCLR1);
+
+#ifdef XMC4_SCU_GATING
+      /* Gate clock */
+
+      putreg32(SCU_CGAT1_CCU43, XMC4_SCU_CGATSET1);
+#endif
+    }
+  else
+    {
+      uint32_t scu_ccu4x_mask = (uint32_t)(1 << (priv->module + 2));
+
+      putreg32(scu_ccu4x_mask, XMC4_SCU_PRSET0);
+
+#ifdef XMC4_SCU_GATING
+      putreg32(scu_ccu4x_mask, XMC4_SCU_CGATSET0);
+#endif
+    }
+
+  pwminfo("PWM CCU4%d module is disabled\n", priv->module);
+
+  return OK;
+}
+
+/*****************************************************************************
+ * Name: pwm_is_module_used
+ *
+ * Description:
+ *   Get the state (running timer or not) of the slice.
+ *
+ * Returned Value:
+ *   0 if module is idle, else true if one slice or more is running.
+ *
+ *****************************************************************************/
+
+static bool pwm_is_module_used(struct xmc4_pwm_s *priv)
+{
+  DEBUGASSERT(priv != NULL);
+
+  uint32_t idle_status = xmc4_pwm_getreg32(priv, XMC4_CCU4_GSTAT_OFFSET);
+  uint32_t mask = (CCU4_GSTAT_S0I_MASK |
+                   CCU4_GSTAT_S1I_MASK |
+                   CCU4_GSTAT_S2I_MASK |
+                   CCU4_GSTAT_S3I_MASK);
+
+  idle_status &= mask;
+
+  bool is_used = !(idle_status == mask);
+
+  pwminfo("PWM CCU4%d is used ? : %d\n", priv->module, is_used);
+
+  return is_used;
+}
+
+/*****************************************************************************
+ * Name: pwm_get_config
+ *
+ * Description:
+ *   Compute the prescaler value, compare and period match for the given
+ *   info config.
+ *
+ * Input Parameters:
+ *   priv - A reference to the lower half PWM driver state structure
+ *   info - A reference to the characteristics of the pulsed output
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure
+ *
+ *****************************************************************************/
+
+static int pwm_get_config(struct xmc4_pwm_s *priv,
```

Comment:
the name is missleading. Maybe pwm_compute_config would be more appropriate ?

Reply:
Changed to pwm_compute_config()